src/daemon/utils.rs

```diff
@@ -0,0 +1,84 @@
+#[cfg(test)]
+pub mod test_utils {
+    use crate::{
+        jsonrpc::UserRole,
+        revaultd::RevaultD,
+        threadmessages::{BitcoindMessageOut, SigFetcherMessageOut},
+        RpcUtils,
+    };
+    use common::config::Config;
+
+    use std::{
+        fs,
+        path::PathBuf,
+        sync::{mpsc, Arc, RwLock},
+        thread,
+    };
+
+    // Create a RevaultD state instance using a scratch data directory, trying to be portable
+    // across UNIX, MacOS, and Windows
+    pub fn dummy_revaultd(datadir_name: &'static str, role: UserRole) -> RevaultD {
+        let repo_root = PathBuf::from(file!())
+            .parent()
+            .unwrap()
+            .parent()
+            .unwrap()
+            .parent()
+            .unwrap()
+            .to_path_buf();
+        let datadir_path: PathBuf = [repo_root.to_str().unwrap(), "test_data", datadir_name]
+            .iter()
+            .collect();
+
+        let config = match role {
+            UserRole::Stakeholder => "stake_config.toml",
+            UserRole::Manager => "man_config.toml",
+            UserRole::ManagerStakeholder => "stake_man_config.toml",
+        };
```

Comment:
How about constructing `RevaultD` structs directly? Configuration files weren't a good idea of mine :/

Reply:
What about pasting the file there as I am doing now instead? It's quicker for me to write, but I agree that it's not that clean

Reply:
Ok for now, can you leave a TODO?